iocore/aio/AIO.cc

```diff
@@ -573,28 +669,15 @@ AIOCallbackInternal::handle_complete(io_uring_cqe *cqe)
     }
   }
 }
+
 #endif
 
 int
 ink_aio_read(AIOCallback *op_in, int fromAPI)
 {
 #if TS_USE_LINUX_IO_URING
   if (use_io_uring) {
-    IOUringContext *ur      = IOUringContext::local_context();
-    AIOCallbackInternal *op = static_cast<AIOCallbackInternal *>(op_in);
-    while (op) {
-      op->this_op       = op;
-      io_uring_sqe *sqe = ur->next_sqe(op);
-      io_uring_prep_read(sqe, op->aiocb.aio_fildes, op->aiocb.aio_buf, op->aiocb.aio_nbytes, op->aiocb.aio_offset);
-      op->aiocb.aio_lio_opcode = LIO_READ;
-      if (op->then) {
-        sqe->flags |= IOSQE_IO_LINK;
-      } else {
-        op->aio_op = static_cast<AIOCallbackInternal *>(op_in);
-      }
-
-      op = static_cast<AIOCallbackInternal *>(op->then);
-    }
+    io_uring_prep_ops_internal(static_cast<AIOCallbackInternal *>(op_in), LIO_READ);
```

Comment:
I would push the static_cast into io_uring_prep_ops_internal.

Reply:
Sometimes this is called with a pointer type that is already AIOCallbackInternal.  I'm not a fan of these casts at all, but they are here to coexist with the thread AIO mode.